src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -83,10 +80,11 @@ public boolean isIllegalReflectionLog(String logString) {
      * @return boolean indicating an unnecessary build log or not
      */
     public boolean isUnnecessaryBuildLogForProgrammingLanguage(String logString, ProgrammingLanguage programmingLanguage) {
-        boolean isInfoWarningOrErrorLog = isInfoLog(logString) || isWarningLog(logString) || isErrorLog(logString) || isDockerImageLog(logString);
+        boolean isInfoWarningOrErrorLog = isInfoLog(logString) || isWarningLog(logString) || isDockerImageLog(logString) || isGitLog(logString) || isTaskLog(logString);
         if (ProgrammingLanguage.JAVA.equals(programmingLanguage)) {
-            return isInfoWarningOrErrorLog || logString.startsWith("Unable to publish artifact") || logString.startsWith("NOTE: Picked up JDK_JAVA_OPTIONS")
-                    || logString.startsWith("Picked up JAVA_TOOL_OPTIONS") || logString.startsWith("[withMaven]") || logString.startsWith("$ docker");
+            return isInfoWarningOrErrorLog || isMavenErrorLog(logString) || isGradleErrorLog(logString) || isGradleInfoLog(logString)
+                    || logString.startsWith("NOTE: Picked up JDK_JAVA_OPTIONS") || logString.startsWith("Picked up JAVA_TOOL_OPTIONS") || logString.startsWith("[withMaven]")
+                    || logString.startsWith("$ docker");
```

Comment:
```suggestion
                    || isCiSystemInfoLog(logString);
```
Maybe they can be grouped together somehow as well. Even something like `startsWithPrefixThatShouldBeIgnored(logString)` if no common element can be found would shorten the code here.